turbopack/crates/turbopack-node/js/src/ipc/index.ts

```diff
@@ -40,7 +41,31 @@ export type Ipc<TIncoming, TOutgoing> = {
 function createIpc<TIncoming, TOutgoing>(
   port: number
 ): Ipc<TIncoming, TOutgoing> {
-  const socket = createConnection(port, "127.0.0.1");
+  const socket = createConnection({
+    port,
+    host: "127.0.0.1",
+    noDelay: true,
+  });
+
+  /**
+   * A writable stream that writes to the socket.
+   * We don't write directly to the socket because we need to
+   * handle backpressure and wait for the socket to be drained
+   * before writing more data.
+   */
+  const socketWritable = new Writable({
+    write(chunk, _enc, cb) {
+      if (socket.write(chunk)) {
+        cb();
+      } else {
+        socket.once('drain', cb);
+      }
+    },
+    final(cb) {
```

Comment:
when is `final` called?  when we call `end` on the socketWritable?

Reply:
Yeah as far as I read the docs/types this is the interface for writeable for cleanup when it ends. But the socket is never cleaned in the first place from what I found in the ipc code.